src/cli/cmd/mod.rs

```diff
@@ -367,6 +382,99 @@ macro_rules! flakehub_url {
     }};
 }
 
+fn is_root_user() -> bool {
+    nix::unistd::getuid().is_root()
+}
+
+async fn nix_command(args: &[&str], sudo_if_necessary: bool) -> Result<(), FhError> {
+    command_exists("nix")?;
+
+    let use_sudo = sudo_if_necessary && !is_root_user();
+
+    let mut cmd = if use_sudo {
+        tracing::warn!(
+            "Current user is {} rather than root; running Nix command using sudo",
+            whoami::username()
+        );
+        tracing::debug!(
+            "Running: sudo nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
+
+        let mut cmd = tokio::process::Command::new("sudo");
```

Comment:
I wonder if we'll run into issues where the user's sudo config does / doesn't forward some environment variables.

Something to be aware of, I guess. (Maybe we should allow the `sudo` command to be customized, so that users have an escape hatch.... but I guess we can revisit that if/when people have issues.)

Reply:
Let's revisit on error.